app/src/pages/[resourceType]/[id].tsx

```diff
@@ -80,7 +80,7 @@ export default function ResourceIDPage({ jsonData }: InferGetServerSidePropsType
 
   useEffect(() => {
     if (dataRequirements?.relatedArtifact) {
-      setSortedDependencies(sortDependencies(dataRequirements.relatedArtifact));
+      setSortedDependencies(sortDependencies(dataRequirements.relatedArtifact.filter(r => r.resource)));
```

Comment:
I moved the check for a defined `resource` field to the DependencyCards component in https://github.com/projecttacoma/measure-repository/pull/64/commits/d3f36378864f826d6e72d8e05a2c8f1d21e87c68 - I think this is a better option so that we have all the dependency card logic contained in the single component.

In terms of these related artifacts not appearing in the app, I think concrete tasking should be defined if we want to make any changes to what fields are checked/rendered in the dependency cards component. My goal was to add more type checking to fit the current functionality. Open to feedback though! The measure I tested with generates data requirements with related artifacts that are of the format `{type: 'depends-on', url: <ValueSet-url>}`, so I'm not sure how helpful it would be to put that info into a dependency card

Reply:
I think I am just confused about what exactly we want in the "Dependencies" tab...I was under the impression that we wanted to list what was in the `relatedArtifacts` of the `dataRequirements` output Library there? I could be wrong though, I just thought that tab was to just display that part of the output of dataRequirements and in that case we should show all of the entries. @hossenlopp what do you think? If we choose to not include all of the relatedArtifacts, then the only thing that should change is the dependency count (currently includes all of them): 

![image](https://github.com/projecttacoma/measure-repository/assets/30158156/ad93d6dd-206f-4eb2-afb7-2c76df15d2aa)

Reply:
I'm also under the impression that we want to show all the entries, but the current functionality on the main branch shows empty cards because they do not have the `resource` field defined, which is required under our current implementation of the dependency cards:
![Screenshot 2023-11-01 at 12 23 28 PM](https://github.com/projecttacoma/measure-repository/assets/87094479/ff05e6f9-06a0-45a7-b042-a62d622be787)
As a quick fix, I added handling to check that the `resource` and `display` fields are defined as a way of additional type handling, under the impression that larger changes to what fields are rendered in a dependency card would require a more explicit backlog task. But we could make the change here now if that seems more appropriate. Here's an example of displaying either the `resource` or `url` field:
![Screenshot 2023-11-01 at 12 24 04 PM](https://github.com/projecttacoma/measure-repository/assets/87094479/086f8ffb-0cfd-4524-a213-dc348ae402be)

Reply:
I like this fix for now! I think it makes sense to list them all regardless of what is currently "correct" since a lot of measures still use `url` instead of `resource`. Absolutely open to more thoughts on this, but I think this is good!

Reply:
Added this fix in https://github.com/projecttacoma/measure-repository/pull/64/commits/d80357da5214f844be338df4ea69b2543f70c2e0

A new issue that I found is that the dependency cards may link to a resource on the measure repository service, but the link is constructed using the url on the related artifact. This url may not be the same as the id of the resource (especially when we are assigning uuid's). Example: trying to route to `/Library/AdultOutpatientEncounters` because that is a related artifact, but the id of the AdultOutpatientEncounters resource is a uuid. This throws an OperationOutcome because the resource can't be found.

I added a TODO comment where this logic occurs. Not sure if this should be made into a backlog task. I couldn't find an easy fix here (requires looking at the related artifact and mapping its url to its id in order to derive the correct routing path). Not sure if we can make the assumption yet that the url is unique either, which might make this tough

Reply:
I agree, I think this should be a backlog task. In fact, it could be worth making a general dependency fix task to look into this more. But I think this is out of scope for this task so I think this is good for now!